Detector/DetFCChhTrackerSimple/src/GenericBarrel_Geo.cpp

```diff
@@ -0,0 +1,161 @@
+
+#include "DetCommon/DetUtils.h"
+
+#include "DD4hep/DetFactoryHelper.h"
+
+using DD4hep::Geometry::Volume;
+using DD4hep::Geometry::DetElement;
+using DD4hep::XML::Dimension;
+using DD4hep::Geometry::PlacedVolume;
+
+namespace det {
+
+static DD4hep::Geometry::Ref_t createGenericTrackerBarrel(DD4hep::Geometry::LCDD& lcdd,
+                                                          DD4hep::XML::Handle_t xmlElement,
+                                                          DD4hep::Geometry::SensitiveDetector sensDet) {
+  // shorthands
+  DD4hep::XML::DetElement xmlDet = static_cast<DD4hep::XML::DetElement>(xmlElement);
+  Dimension dimensions(xmlDet.dimensions());
+  // get sensitive detector type from xml
+  DD4hep::XML::Dimension sdTyp = xmlElement.child("sensitive");
+  if (xmlDet.isSensitive()) {
+    // sensitive detector used for all sensitive parts of this detector
+    sensDet.setType(sdTyp.typeStr());
+  }
+  // definition of top volume
+  // has min/max dimensions of tracker for visualization etc.
+  std::string detectorName = xmlDet.nameStr();
+  DetElement topDetElement(detectorName, xmlDet.id());
+  DD4hep::Geometry::Tube topVolumeShape(dimensions.rmin(), dimensions.rmax(), dimensions.dz());
+  Volume topVolume(detectorName, topVolumeShape, lcdd.air());
+  topVolume.setVisAttributes(lcdd.invisible());
+
+  // counts all layers - incremented in the inner loop over repeat - tags
+  unsigned int idxLay = 0;
+  // loop over 'layer' nodes in xml
+  for (DD4hep::XML::Collection_t xLayerColl(xmlElement, _U(layers)); nullptr != xLayerColl; ++xLayerColl) {
+    DD4hep::XML::Component xLayer = static_cast<DD4hep::XML::Component>(xLayerColl);
+
+    DD4hep::XML::Component xModuleComponents = xmlElement.child("module_components");
+    DD4hep::XML::Component xModule =
+        det::getNodeByStrAttr(xmlElement, "module", "name", xLayer.attr<std::string>("module"));
+
+    // optional parameters
+    double stereo_offset = det::getDoubleWithDefault(xLayer, "stereo_offset", 0.0);
+    double module_twist_angle = det::getDoubleWithDefault(xLayer, "module_twist_angle", 0.1 * M_PI);
+    double stereo_module_overlap = det::getDoubleWithDefault(xLayer, "stereo_module_overlap", 0.0);
```

Comment:
Can you add comments in the xml that you have these hardcoded defaults if they are not specified in the xml?